optimization/univariate/un.metta

```diff
@@ -0,0 +1,87 @@
+;; - Repeat for limited generations and return best center with updated Deme
+;;
+;; Inputs
+;; - Deme contains Representation with knob map and tree, and an InstanceSet possibly scored
+;; - TruthTableBScore carries complexity coefficient and input table for scoring
+;; - Instance is the initial center from which neighborhood exploration starts
+;;
+;; Output
+;; - Returns an Instance, a Deme with scored instances, and a state tuple
+
+;; Entry point compatible with runMoses/expandDeme (same signature as hillClimbing)
+; (: univariate (-> Deme (TruthTableBScore $a) Instance (Instance Deme $state)))
+(= (univariate $deme $tTableBscorer $centerInst)
+   (let*
+       (
+         (() (println! (Entering univariate with center: $centerInst)))
+         ((mkTruthTableBScore $cpxCoeff $size $iTable) $tTableBscorer)
+         ((mkDeme $rep (mkSInstSet $instSet) $id) $deme)
+         ($discMap (getDiscMap $rep))
+         ($discMapLength (MultiMap.length $discMap))
+         ($maxGens (max 1 $discMapLength))
+         (() (println! (Max generations: $maxGens)))
+       )
+       (univariateLoop $deme $cpxCoeff $iTable $centerInst 0 $maxGens)
+   ))
+
+; (: univariateLoop (-> Deme Number (ITable $a) Instance Number Number (Instance Deme $state)))
+(= (univariateLoop (mkDeme $rep (mkSInstSet $instSet) $id)
+                    $cpxCoeff $iTable $centerInst $gen $maxGens)
+   (if (>= $gen $maxGens)
+       (let* (
+         (() (println! (Reached max generations: $gen)))
+         ($result ($centerInst (mkDeme $rep (mkSInstSet $instSet) $id) ()))
+       )
+       $result)
+       (let* (
+             (() (println! (Univariate loop gen: $gen)))
+             ($totalNeighbors (estimateNeighborhood 1 (getDiscMap $rep)))
+             ($nNew (min $totalNeighbors 3))
+             (($demeWithNewInsts $added) (sampleNewInstances $totalNeighbors $nNew $centerInst (mkDeme $rep (mkSInstSet $instSet) $id) 1))
+             ((mkSInstSet $scored) (transform (getInstanceSet $demeWithNewInsts) (getRep $demeWithNewInsts) $iTable $cpxCoeff))
+             ($demeScored (mkDeme (getRep $demeWithNewInsts) (mkSInstSet $scored) (getDemeId $demeWithNewInsts)))
+             ((mkSInst $bestPair) (List.foldl ((curry2 returnBest) <) (mkSInst (mkPair $centerInst (worstCscore))) $scored))
+             (() (println! (returnBest returned: $bestPair)))
+             ($nextCenter (Pair.first $bestPair))
+             (() (println! (Next center: $nextCenter)))
+           )
+           (univariateLoop $demeScored $cpxCoeff $iTable $nextCenter (+ $gen 1) $maxGens)
+        )
+       )
+   )
+
+;; Helper function to extract the discrete map from a representation
+(: getDiscMap (-> Representation (MultiMap (DiscSpec $knob))))
+(= (getDiscMap (mkRep (mkKbMap $dscKbMp $dscMp) $tree)) (extractMultiMap $dscMp))
+
+;; Helper function to extract MultiMap from DiscMap
+(: extractMultiMap (-> DiscMap (MultiMap (DiscSpec $knob))))
+(= (extractMultiMap (mkDscMp $multiMap)) $multiMap)
```

Comment:
This is also a `DiscMap` helper but it's defintion is found within the `representation.metta` file so best to keep it there